src/Modules/Profile/Commands.Profile/InvokePowerBIRestMethod.cs

```diff
@@ -0,0 +1,108 @@
+
+        [Parameter(Mandatory = false)]
+        public string Organization = "myorg";
+
+        [Parameter(Mandatory = false)]
+        public string Version = "v1.0";
+        #endregion
+
+        public override void ExecuteCmdlet()
+        {
+            if(Uri.TryCreate(this.Url, UriKind.Absolute, out Uri testUri))
+            {
+                this.Url = testUri.AbsoluteUri;
+            }
+            else
+            {
+                this.Url = $"{this.Version}/{this.Organization}/" + this.Url;
+            }
+            
+            var response = this.InvokeRestMethod(this.Url, this.Body, this.Method).Result;
+            
+            var result = response.Content.ReadAsStringAsync().Result;
+            if (result != null)
+            {
+                this.Logger.WriteObject(result);
+            }
+        }
+
+        private async Task<HttpResponseMessage> InvokeRestMethod(string url, string body, PowerBIWebRequestMethod requestType)
+        {
+            // https://msdn.microsoft.com/en-us/library/mt243842.aspx
+            // https://docs.microsoft.com/en-us/aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client
+            var token = this.Authenticator.Authenticate(this.Profile, this.Logger, this.Settings);
+            using (var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri(this.Profile.Environment.GlobalServiceEndpoint);
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+                client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", token.AccessToken);
+
+                HttpResponseMessage response = null;
+                switch (requestType)
+                {
+                    case PowerBIWebRequestMethod.Get:
+                        response = await client.GetAsync(url);
+                        break;
+                    case PowerBIWebRequestMethod.Post:
+                        response = await client.PostAsync(url, new StringContent(body));
+                        break;
+                    case PowerBIWebRequestMethod.Delete:
+                        response = await client.DeleteAsync(url);
+                        break;
+                    case PowerBIWebRequestMethod.Put:
+                        response = await client.PutAsync(url, new StringContent(body));
+                        break;
+                    case PowerBIWebRequestMethod.Patch:
+                        response = await client.SendAsync(new HttpRequestMessage(new HttpMethod("PATCH"), url) { Content = new StringContent(body) });
+                        break;
+                    case PowerBIWebRequestMethod.Options:
+                        response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Options, url));
```

Comment:
You could consider making HttpClient extension methods for Patch and Options.

Reply:
Good idea, we can do that in a future update